packages/components/src/TooltipTrigger/TooltipTrigger.component.js

```diff
@@ -22,17 +24,75 @@ const props = {
 	<Icon name="my-icon" />
 </TooltipTrigger>
  */
-function TooltipTrigger(props) {
-	const tooltip = (<Tooltip className={getTooltipClass()} id={uuid.v4()}>{props.label}</Tooltip>);
-	return (
-		<OverlayTrigger
-			placement={props.tooltipPlacement}
-			overlay={tooltip}
-			delayShow={400}
-		>
-			{props.children}
-		</OverlayTrigger>
-	);
+class TooltipTrigger extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.onMouseOver = this.onMouseOver.bind(this);
+		this.onMouseOut = this.onMouseOut.bind(this);
+		this.state = {
+			hovered: false,
+			id: uuid.v4(),
+		};
+	}
+
+	shouldComponentUpdate(nextProps, nextState) {
+		if (nextState.hovered !== this.state.hovered) {
+			return true;
+		}
+		if (this.props.children !== nextProps.children) {
+			return true;
+		}
+		if (this.props.label !== nextProps.label) {
+			return true;
+		}
+		return false;
+	}
+
+	onMouseOver() {
+		this.setState({ hovered: true });
```

Comment:
think this should check actual state, to avoid triggering a render call and even avoid calling shouldComponentUpdate